Assets/Scripts/Framework/Statemachine/SmallEnemiesStateMachine.cs

```diff
@@ -1,4 +1,7 @@
 public class SmallEnemiesStateMachine : FiniteStateMachine
 {
+    public Idle idleState;
+    public Walking walkingState;
```

Comment:
```suggestion
    [HideInInspector] public Idle idleState;
    [HideInInspector] public Walking walkingState;
```
Ze zou dit kunnen doen en dan in de awake een GetComponent<>() kunnen maken voor elke state.

Reply:
@bas-boop just so you know, GetComponent calls zouden niet nodig zijn met een StateToStateMachine Subscription relatie :p

Reply:
Met wat Mark hierboven heb staan, zou je het waarschijnlijk beter op deze manier kunnen typen.

Je kan nog steeds nu in de inspector dingen slepen. Je kan in andere script nu niks Setten, maar wel de SwitchState() gebruiken en een state mee geven.

```suggestion
    [field: SerializeField] public Idle idleState { get; private set; }
    [field: SerializeField] public Walking walkingState { get; private set; }
```